code/game/dna/dna_modifier.dm

```diff
@@ -589,6 +589,8 @@
 		var/lock_state = src.connected.locked
 		src.connected.locked = 1//lock it
 
+		nanomanager.update_uis(src) // show the splash while this works
```

Comment:
Yeah these aren't good, dna scanner is one of the laggiest nanouis and though I'm fairly sure manual refreshes were removed intentionally, I'm definitely sure there shouldn't be any.